crates/services/upgradable-executor/wasm-executor/src/main.rs

```diff
@@ -114,15 +115,16 @@ fn execute_validation(
     block: Block,
 ) -> ExecutorResult<Uncommitted<ExecutionResult, Changes>> {
     let (
-        ExecutionResult {
-            block,
+        ValidationResult {
             skipped_transactions,
             tx_status,
             events,
         },
         changes,
-    ) = instance.validate_without_commit(block)?.into();
+    ) = instance.validate_without_commit(&block)?.into();
 
+    // We have to return the same type here because if the input parsing fails it won't know if this
+    // is a validation or production block.
```

Comment:
Maybe instead of serializing `ReturnType::V1(result)` in `extern "C" fn execute`, we serialize some `Result<ReturnType, Error>`?

That way we don't need the `ReturnType` variant to return an `Err`.

i.e. We can do something like:
```
enum ReturnType {
    V1(V1ReturnType)
}

enum V1ReturnType {
    Execute(ExecutionResult),
    Validate(ValidationResult),
}
```

Reply:
We want to return an `ExecutorResult` because we may change the error type later. We could add ` V2` that returns a validation result and name it `Validation` instead of `V2`=)

But I don't see any problems in returning `ExecutionResult` with a `block` type since we don't consume `block`, and it doesn't hurt the performance.

Reply:
Actually, it affects performance. We need to serialize and deserialize the block on the upgradeable executor side, so maybe we should add a new return variant. But let’s do it in a separate PR, we need to add an integration test for it to be sure that we did everything correct=)

Reply:
https://github.com/FuelLabs/fuel-core/issues/1887